### ISSUE TITLE
KeyError: 0

### ISSUE THREAD

**Issue by niftylettuce** [ValiMail/authentication-headers#9]
There seems to be a bug when trying to parse DKIM...  It's trying to use 0th key but it doesn't exist, and thus throws this error.

```sh
Traceback (most recent call last):␊
      File "/Users/user/Projects/authheaders/scripts/authenticate-message.py", line 33, in <module>␊
        main()␊
      File "/Users/user/Projects/authheaders/scripts/authenticate-message.py", line 24, in main␊
        header = authheaders.authenticate_message(msg=message, authserv_id=authservId, ip=ip, mail_from=mailFrom, helo=helo, spf=True, dkim=True, arc=True)␊
      File "/usr/local/lib/python3.8/site-packages/authheaders/__init__.py", line 282, in authenticate_message␊
        dkim_result = check_dkim(msg)␊
      File "/usr/local/lib/python3.8/site-packages/authheaders/__init__.py", line 63, in check_dkim␊
        res = d.verify() and 'pass' or 'fail'␊
      File "/usr/local/lib/python3.8/site-packages/dkim/__init__.py", line 940, in verify␊
        return self.verify_sig(sig, include_headers, sigheaders[idx], dnsfunc)␊
      File "/usr/local/lib/python3.8/site-packages/dkim/__init__.py", line 773, in verify_sig␊
        self.pk, self.keysize, self.ktag, self.seqtlsrpt = load_pk_from_dns(name,␊
      File "/usr/local/lib/python3.8/site-packages/dkim/__init__.py", line 481, in load_pk_from_dns␊
        s = dnsfunc(name, timeout=timeout)␊
      File "/usr/local/lib/python3.8/site-packages/dkim/dnsplug.py", line 88, in get_txt␊
        txt = _get_txt(unicode_name, timeout)␊
      File "/usr/local/lib/python3.8/site-packages/dkim/dnsplug.py", line 34, in get_txt_dnspython␊
        return b"".join(r.items[0].strings)␊
    KeyError: 0
```

**Comment by niftylettuce** [comment]
This seems directly related to #8 in that dnspython==2.1.0.dev1 is not compatible with this yet.

This is literally a game of whackamole at this point.

**Comment by niftylettuce** [comment]
This seems to be a dkimpy issue, so looping in @kitterma here.

**Comment by niftylettuce** [comment]
Actually, the fix is to use `dnspython==1.16.0`, the latest version of dnspython is not compatible with this yet.

**Comment by niftylettuce** [comment]
Re-opening since this is a known bug in dkimpy with a resolution already documented

https://bugs.launchpad.net/dkimpy/+bug/1888583

**Comment by kitterma** [comment]
No.  It's not an authheader's issue, so it's not appriopriate here.  You can install the older dnspython or use pydns with dkimpy to work around it.

**Comment by kitterma** [comment]
dkimpy 1.0.5 has been released which should work with dnspython 2.0.0.  You can upgrade both to the latest now.
